specification/kubernetesruntime/KubernetesRuntime.Management/storageclass.tsp

```diff
@@ -0,0 +1,276 @@
+  NFS,
+
+  @doc("SMB storage class")
+  SMB,
+}
+
+@doc("Data resilience tier of a storage class")
+enum DataResilienceTier {
+  @doc("Not data resilient")
+  NotDataResilient,
+
+  @doc("Data resilient")
+  DataResilient,
+}
+
+@doc("Ability to expand volumes of a storage class")
+enum VolumeExpansion {
+  @doc("Allow volume expansion")
+  Allow,
+
+  @doc("Disallow volume expansion")
+  Disallow,
+}
+
+@doc("Storage Class Access Mode")
+enum AccessMode {
+  @doc("Read Write Once (RWO) access mode")
+  ReadWriteOnce,
+
+  @doc("Read Write Many (RWX) access mode")
+  ReadWriteMany,
+}
+
+@doc("Storage class volume binding mode")
+enum VolumeBindingMode {
+  @doc("Immediate binding mode")
+  Immediate,
+
+  @doc("Wait for first consumer binding mode")
+  WaitForFirstConsumer,
+}
+
+@doc("Details of the StorageClass StorageClass.")
+model StorageClassProperties {
+  @doc("Volume can be expanded or not")
+  allowVolumeExpansion?: VolumeExpansion;
+
+  @doc("Additional mount options")
+  mountOptions?: string[];
+
+  @doc("Provisioner name")
+  @visibility("read", "create")
+  provisioner?: string;
+
+  @doc("Binding mode of volumes: Immediate, WaitForFirstConsumer")
+  @visibility("read", "create")
+  volumeBindingMode?: VolumeBindingMode;
+
+  @doc("The access mode: [ReadWriteOnce, ReadWriteMany] or [ReadWriteOnce]")
+  accessModes?: AccessMode[];
+
+  @doc("Allow single data node failure")
+  dataResilience?: DataResilienceTier;
+
+  @doc("Failover speed: NA, Slow, Fast")
+  failoverSpeed?: FailoverTier;
+
+  @doc("Limitations of the storage class")
+  limitations?: string[];
+
+  @doc("Performance tier")
+  performance?: PerformanceTier;
+
+  @doc("Selection priority when multiple storage classes meet the criteria. 0: Highest, -1: Never use")
+  priority?: int64;
+
+  @doc("Properties of the StorageClass")
+  // prevent generation of buggy StorageClassPropertiesUpdate model
+  @visibility("read", "create")
+  typeProperties: StorageClassTypeProperties;
+
+  @doc("Resource provision state")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@discriminator("type")
+@doc("The properties of storage class of the StorageClass")
+model StorageClassTypeProperties {
+  @doc("Type of the storage class.")
+  type: SCType;
+}
+
+@doc("The properties of Native StorageClass")
+model NativeStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("Native StorageClass")
+  type: SCType.Native;
+}
+
+@doc("The properties of RWX StorageClass")
+model RwxStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("RWX StorageClass")
+  type: SCType.RWX;
+
+  @doc("The backing storageclass used to create new storageclass")
+  backingStorageClassName: string;
+}
+
+@doc("The properties of Blob StorageClass")
+model BlobStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("Blob StorageClass")
+  type: SCType.Blob;
+
+  @doc("Azure Storage Account Name")
+  azureStorageAccountName: string;
```

Comment:
>azureStorageAccountName [](http://example.com/codeflow?start=2&length=23)

Is a name enough? What if its not on the main storage DNS but on partitioned DNS? Or custom domain? Can you provide full endpoint / endpoint suffix?

Reply:
We are using [blob-csi-driver](https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/docs/driver-parameters.md) for storage classes of this SCType. `azureStorageAccountName` is the only information it requires.